inst/include/dust/dust.hpp

```diff
@@ -290,10 +290,11 @@ template <typename T>
 std::vector<typename T::real_t>
 dust_simulate(const std::vector<size_t> steps,
               const std::vector<typename T::init_t> data,
-              const std::vector<typename T::real_t> state,
+              std::vector<typename T::real_t>& state,
```

Comment:
Should data (on the line above) also be passed by reference. I got stung by some of these being missing when I scaled up to 100k particles, causing endless copies of enormous vectors!

Reply:
good shout